RockWeb/Blocks/Communication/EmailPreferenceEntry.ascx.cs

```diff
@@ -227,11 +228,27 @@ protected override void OnLoad( EventArgs e )
         /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
         protected void btnSubmit_Click( object sender, EventArgs e )
         {
+            var rockContext = new RockContext();
             nbUnsubscribeSuccessMessage.Visible = false;
             nbEmailPreferenceSuccessMessage.Visible = false;
 
             if ( rbUnsubscribe.Checked && rbUnsubscribe.Visible )
             {
+                Guid? workflowGuid = GetAttributeValue( "UnsubscribeWorkflow" ).AsGuidOrNull();
+                WorkflowTypeCache workflowType = null;
+                var workflowService = new WorkflowService( rockContext );
+
+                if (workflowGuid != null)
+                {
+                    workflowType = WorkflowTypeCache.Get( workflowGuid.Value );
+                }
+
+                // Start workflow for this person
+                if (workflowType != null)
+                {
+                    Dictionary<string, string> attributes = new Dictionary<string, string>();
```

Comment:
Hi @engineeringdrew I see this `attributes` list you pass to the `StartWorkflow` method _would_ be used by setting them into the launched workflow (`workflow.SetAttributeValue(...)`) but I don't see you populating this list anywhere.  Did you miss this? (Or am I missing something?)

Reply:
I just left that in there in case any developer wanted to use it.  Right now it doesn't do anything.

Reply:
@engineeringdrew I guess what I should have said is -- you're passing in a completely _empty_ dictionary.  So the `foreach` loop in the `StartWorkflow` method will never do anything.  It's just extra code and should be removed -- because otherwise, glancing at the `StartWorkflow` method, a developer might not even realize there are _no_ attributes in the dictionary.

On the other hand, if you know there's something in the block's context-data that should be passed in (for an example, see https://github.com/SparkDevNetwork/Rock/commit/ae912c29e44ef7c9df0428a4b01ad3f8b037c5d2#diff-589989fd7f3358571965f779a866aaf3R381) then it could make sense to add them to the dictionary and leave the code in the StartWorkflow method as is.

Let me know what you think about this.

Thanks for looking at this again -- and for helping make Rock better!